### DISCUSSION TITLE
symlink-content-astro-portability/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#611] [Comments]
# symlink-content-astro-portability/

Azure / Microsoft 365 / Visual Studio Code / Speaker / Blogger

https://www.eliostruyf.com/symlink-content-astro-portability/

**Comment by rowemoore** [comment]
I just set up an astro 2.0 project and not having much luck using Front Matter for VSCode. I've tried both the beta and the latest official and neither is able to detect the markdown fields correctly.  I don't have any existing content so I'm guessing I need to do something different than what you have described here?

**Reply by estruyf** [reply]
If you have a repo to share or you can add me to it as a contributor, I'm happy to take a look at it. 

By the way, do you want also use symlinked content?

**Reply by rowemoore** [reply]
I just used - [a clone of astro paper](https://github.com/rowemoore/williwalli-paper.git).  I'm on windows.  I have set up my connection to firebase and have no trouble pushing new content.  Just would like an easier way to create it.  I don't need a symlink to existing content - I don't have any.  
I've added you to my repo.  Thanks!

**Reply by estruyf** [reply]
Thanks @rowemoore, I have just added the Front Matter CMS configuration. All you had to do was add the right page folder path and it started working.

BTW: It is best to use the beta version of Front Matter CMS for now, as it got a fix for Windows where the YAML parser sometimes adds an extra carriage return at the end and makes the parser fail. Beta can be found here: https://marketplace.visualstudio.com/items?itemName=eliostruyf.vscode-front-matter-beta - make sure to disable or uninstall the stable version of the extension. Both cannot run at the same time.

**Reply by rowemoore** [reply]
Thank you!

**Comment by DonKirkham** [comment]
Elio, this is very interesting. I also use Hugo, so I am wondering how this affects your original Hugo project. Are you also symlinking your content from Hugo? I am looking to redesign my website and this approach might allow me to create new Hugo projects where I can work on the redesign while continuing to update my original site.  Exploring other other static website options would be a bonus!

**Reply by estruyf** [reply]
Thank you @DonKirkham for the coffee! 🙏 I will try to write down the process of how I manage my Hugo blog next week. For Hugo it is even easier than symlinking, as it has a configuration option for this.

**Reply by estruyf** [reply]
@DonKirkham, as promised, I wrote down how I manage my Hugo website. You can read more about it here: https://www.eliostruyf.com/managing-hugo-website-content-asset-submodule/

**Comment by DonKirkham** [comment]
Thanks Elio. I need to move my Hugo site to GitHub actions first, and then learn about FrontMatterCMS. Once that is done, I will definitely give this approach a try.

Don
